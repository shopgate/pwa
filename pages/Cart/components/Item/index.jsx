import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import Portal from '@shopgate/pwa-common/components/Portal';
import {
  CART_ITEM_TYPE_PRODUCT,
  CART_ITEM_TYPE_COUPON,
} from '@shopgate/pwa-common-commerce/cart/constants';
import {
  CART_ITEM,
  CART_ITEM_BEFORE,
  CART_ITEM_AFTER,
} from '@shopgate/pwa-common-commerce/cart/constants/Portals';
import Product from './components/Product';
import Coupon from './components/Coupon';

/**
 * The CartItem component.
 * @param {Object} props The component props.
 * @param {Object} props.item The cart item.
 * @param {Function} props.onFocus A function to indicate when the item has been focussed.
 * @return {JSX}
 */
const CartItem = ({ item, onFocus }) => {
  if (item.type !== CART_ITEM_TYPE_PRODUCT && item.type !== CART_ITEM_TYPE_COUPON) {
    return null;
  }

  const props = { item };
  const isProduct = (item.type === CART_ITEM_TYPE_PRODUCT);

  return (
    <Fragment>
      <Portal name={CART_ITEM_BEFORE} props={props} />
      <Portal name={CART_ITEM} props={props}>
        {isProduct ? (
          <Product
            id={item.id}
            key={item.id}
            product={item.product}
            quantity={item.quantity}
            messages={item.messages}
            onToggleFocus={onFocus}
          />
        ) : (
          <Coupon
            id={item.id}
            key={item.id}
            coupon={item.coupon}
            messages={item.messages}
          />
        )}
      </Portal>
      <Portal name={CART_ITEM_AFTER} props={props} />
    </Fragment>
  );
};

CartItem.propTypes = {
  item: PropTypes.shape().isRequired,
  onFocus: PropTypes.func.isRequired,
};

export default CartItem;
