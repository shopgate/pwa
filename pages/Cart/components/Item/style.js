import { css } from 'glamor';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const { colors, variables } = themeConfig;

const messagesContainer = css({
  background: colors.light,
  padding: `${variables.gap.big}px ${variables.gap.big}px 0`,
});

const messages = css({
  borderRadius: 4,
  padding: `${variables.gap.small / 2}px ${variables.gap.small}px`,
  lineHeight: 1.125,
});

export default {
  messagesContainer,
  messages,
};
