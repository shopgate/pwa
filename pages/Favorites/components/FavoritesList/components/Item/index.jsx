import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { findDOMNode } from 'react-dom';
import Transition from 'react-transition-group/Transition';
import Grid from '@shopgate/pwa-common/components/Grid';
import { getAbsoluteHeight } from '@shopgate/pwa-common/helpers/dom';
import CardItem from '@shopgate/pwa-ui-shared/CardList/components/Item';
import CTAButtons from './components/CTAButtons';
import Image from './components/Image';
import ProductInfo from './components/ProductInfo';
import styles from './style';

/**
 * Renders Favorites list item.
 * @param {Object} product Product.
 * @returns {XML}
 */
class Item extends Component {
  static propTypes = {
    product: PropTypes.shape().isRequired,
  };

  /**
   * Constructor.
   * @param {Object} props The component props.
   */
  constructor(props) {
    super(props);
    this.state = {
      visible: true,
    };
  }

  /**
   * Component did update callback.
   */
  componentDidUpdate() {
    this.height = this.getHeight();
  }

  /**
   * Measures height.
   * @returns {number}
   */
  getHeight = () => {
    if (!this.refElement) {
      return 0;
    }
    // eslint-disable-next-line react/no-find-dom-node
    return getAbsoluteHeight(findDOMNode(this.refElement));
  };

  /**
   * Get the element height to determine the translate distance
   * @param {Object} element Component ref
   */
  saveRef = (element) => {
    if (this.refElement) {
      return;
    }
    this.refElement = element;
  };

  /**
   * Renders favorite list item
   * @returns {XML}
   */
  render() {
    return (
      <Transition
        in={this.state.visible}
        timeout={styles.favItemTransitionDuration}
        key={this.props.product.id}
      >
        {state => (
          <CardItem
            ref={this.saveRef}
            className={
              styles.getFavItemTransitionStyle(state, this.state.visible, this.height)
            }
          >
            <Grid className={styles.row}>
              <Grid.Item className={styles.leftColumn}>
                <Image product={this.props.product} />
                <CTAButtons
                  productId={this.props.product.id}
                  active={this.state.visible}
                  removeThrottle={styles.favItemTransitionDuration + 200}
                  onRippleComplete={(active) => {
                    this.setState({
                      visible: active,
                    });
                  }}
                  favoritesOnce
                />
              </Grid.Item>
              <Grid.Item grow={1} className={styles.rightColumn}>
                <ProductInfo product={this.props.product} />
              </Grid.Item>
            </Grid>
          </CardItem>
        )}
      </Transition>
    );
  }
}

export default Item;
