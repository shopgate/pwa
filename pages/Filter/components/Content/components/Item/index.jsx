import React from 'react';
import PropTypes from 'prop-types';
import styles from './style';

/**
 * @returns {JSX}
 */
const FilterContentItem = ({ children }) => (
  <div className={styles}>
    {children}
  </div>
);

FilterContentItem.propTypes = {
  children: PropTypes.node.isRequired,
};

export default FilterContentItem;
