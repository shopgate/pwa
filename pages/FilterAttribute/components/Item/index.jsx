import React from 'react';
import PropTypes from 'prop-types';
import Grid from '@shopgate/pwa-common/components/Grid';
import RippleButton from '@shopgate/pwa-ui-shared/RippleButton';
import Checkbox from '@shopgate/pwa-ui-shared/Checkbox';
import RadioButton from '@shopgate/pwa-ui-shared/RadioButton';
import styles from './style';

/**
 * The Filter Attribute Item component.
 * @param {Object} props The component props.
 * @return {JSX}
 */
const Item = ({
  checked,
  label,
  onClick,
  singleSelect,
}) => (
  <RippleButton type="plain" onClick={onClick} className={styles.item}>
    <Grid>
      <Grid.Item grow={1} shrink={1} className={styles.labelColumn}>
        <div className={styles.label} data-test-id={label}>
          {label}
        </div>
      </Grid.Item>
      <Grid.Item grow={0} shrink={0}>
        <div className={styles.checkbox}>
          {singleSelect ? <RadioButton checked={checked} /> : <Checkbox checked={checked} /> }
        </div>
      </Grid.Item>
    </Grid>
  </RippleButton>
);

Item.propTypes = {
  checked: PropTypes.bool,
  label: PropTypes.string,
  onClick: PropTypes.func,
  singleSelect: PropTypes.bool,
};

Item.defaultProps = {
  checked: false,
  label: null,
  onClick: () => {},
  singleSelect: false,
};

export default Item;
