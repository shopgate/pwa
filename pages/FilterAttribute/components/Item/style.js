import { css } from 'glamor';
import colors from 'Styles/colors';
import variables from 'Styles/variables';

const item = css({
  background: colors.light,
  marginBottom: variables.gap.small / 2,
  width: '100%',
}).toString();

const baseColumn = {
  overflow: 'hidden',
  padding: '5px 0px',
  textOverflow: 'ellipsis',
};

const labelColumn = css({
  overflow: 'hidden',
}).toString();

const label = css({
  ...baseColumn,
  marginRight: '1em',
}).toString();

const checkbox = css({
  ...baseColumn,
}).toString();

export default {
  item,
  labelColumn,
  label,
  checkbox,
};
