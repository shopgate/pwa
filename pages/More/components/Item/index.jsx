import React from 'react';
import PropTypes from 'prop-types';
import Link from '@shopgate/pwa-common/components/Link';
import I18n from '@shopgate/pwa-common/components/I18n';
import styles from './style';

/**
 * @returns {JSX}
 */
function MoreMenuItem({
  href, label, onClick, testId,
}) {
  if (!href && onClick) {
    return (
      <button className={styles} onClick={onClick} data-test-id={testId}>
        <I18n.Text string={label} />
      </button>
    );
  }

  return (
    <Link className={styles} href={href}>
      <I18n.Text string={label} />
    </Link>
  );
}

MoreMenuItem.propTypes = {
  label: PropTypes.string.isRequired,
  href: PropTypes.string,
  onClick: PropTypes.func,
  testId: PropTypes.string,
};

MoreMenuItem.defaultProps = {
  href: null,
  onClick: null,
  testId: null,
};

export default MoreMenuItem;
