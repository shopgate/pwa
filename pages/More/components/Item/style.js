import { css } from 'glamor';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const { colors } = themeConfig;

// place a thick shadow at the top of the first child and a thin shadow otherwise
export default css({
  '&:first-child': {
    boxShadow: '0 0 0 0',
  },
  boxShadow: `0 -1px 0 0 ${colors.darkGray}`,
  padding: '12px 0',
  'button&': {
    outline: 0,
    textAlign: 'inherit',
    width: '100%',
  },
}).toString();
