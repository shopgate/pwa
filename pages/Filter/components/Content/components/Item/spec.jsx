import React from 'react';
import { shallow } from 'enzyme';
import Item from './index';

describe('Filter: <Item />', () => {
  it('should render', () => {
    const wrapper = shallow((
      <Item>
        <div>Some</div>
      </Item>
    ));

    expect(wrapper).toMatchSnapshot();
  });
});
