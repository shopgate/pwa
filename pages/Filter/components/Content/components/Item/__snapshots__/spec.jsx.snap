// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter: <Item /> should render 1`] = `
<div
  className={
    Object {
      "data-css-qsx8t9": "",
    }
  }
>
  <div>
    Some
  </div>
</div>
`;
